core/src/main/java/com/linecorp/armeria/internal/common/AbstractKeepAliveHandler.java

```diff
@@ -181,6 +189,26 @@ public final void onReadOrWrite() {
             lastConnectionIdleTime = System.nanoTime();
         }
 
+        if (!isServer && !isActive) {
```

Comment:
It should be useful if we also support `ConnectionPoolListener` for servers. Although we are focusing on the client-side implementation, the public API design should take extensibility into consideration. 

I propose to add a new `ConnectionPoolListener` to `c.l.a.common` package. The new methods will be added to the new `ConnectionPoolListener` which extends the old `ConnectionPoolListener` for compatibility. For example:
https://github.com/line/armeria/blob/1d268aa278524997ad5c1cc8f66661cfabbb6dd1/core/src/main/java/com/linecorp/armeria/common/encoding/StreamDecoder.java#L27-L27

Related issue: https://github.com/line/armeria/issues/3722

Reply:
@ikhoon 
I agreed to extend the interface for servers.👍

Should I check the instance type and call the new methods using type casting in client-side implementation?🤔
For example,

```
// com.linecorp.armeria.client.HttpChannelPool -> call connectionActive

try {
  if (listener instanceof com.linecorp.armeria.common.ConnectionPoolListener) {
    ((com.linecorp.armeria.common.ConnectionPoolListener) listener).connectionActive(
      protocol, remoteAddr, localAddr, channel, true);
  } else {
    listener.connectionOpen(protocol, remoteAddr, localAddr, channel);
  }
} catch (Throwable e) {
  if (logger.isWarnEnabled()) {
    logger.warn("{} Exception handling {}.connectionActive()", channel, listener.getClass().getName(), e);
  }
}
```